targets/TARGET_NUVOTON/TARGET_M480/lp_ticker.c

```diff
@@ -163,14 +222,15 @@ const ticker_info_t* lp_ticker_get_info()
 
 static void tmr1_vec(void)
 {
-    TIMER_ClearIntFlag((TIMER_T *) NU_MODBASE(TIMER_MODINIT.modname));
-    wait_us((NU_US_PER_SEC / NU_TMRCLK_PER_SEC) * 3);
-
-    TIMER_ClearWakeupFlag((TIMER_T *) NU_MODBASE(TIMER_MODINIT.modname));
-    wait_us((NU_US_PER_SEC / NU_TMRCLK_PER_SEC) * 3);
+    /* NOTE: We need to clear interrupt flag earlier to reduce possibility of dummy interrupt.
+     * This is because "clear interrupt flag" needs delay which isn't added here to avoid
+     * blocking in ISR code. */
+    lp_ticker_clear_interrupt();
```

Comment:
does this clear take effect immediately, or is the interrupt stuck pending for a period of time (causing tmr1_vec to get called multiple times)?

Reply:
Having tmr1_vec getting stuck pending is a problem as well because lower priority interrupts or threads won't get a chance to run. As an alternative fix, would it be possible to disable the NVIC interrupt in place of setting NU_TICKER_INIT_INTR_DIS and clear+enable in place of NU_TICKER_INIT_INTR_EN? That way the interrupt won't fire repeatedly.

Reply:
@c1728p9 That doesn't help. As `lp_ticker_set_interrupt` (`NVIC_EnableIRQ`) or `lp_ticker_fire_interrupt` (`NVIC_EnableIRQ`) are called in `lp_ticker_irq_handler`, `tmr1_vec` (`NVIC_DisableIRQ`) is still getting stuck. We will be in a loop of `tmr1_vec` > `lp_ticker_fire_interrupt`/`lp_ticker_fire_interrupt` > `tmr1_vec` > `lp_ticker_fire_interrupt`/`lp_ticker_fire_interrupt` until clear flag takes effect.

Reply:
@c1728p9 Correct my comment above. Your idea would be workable. We won't get stuck in a loop of `tmr1_vec` (clear interrupt flag and disable interrupt at the start) and `lp_ticker_fire_interrupt`/`lp_ticker_set_interrupt` (called inside `lp_ticker_irq_handler` and so interrupt re-enabled) before the above "clear interrupt flag" takes effect:
1. When `lp_ticker_fire_interrupt` is called, it means there is past event and so the interrupt with uncompleted "clear interrupt flag" gets not dummy.
1. With `LPTICKER_DELAY_TICKS` enabled, it is `lp_ticker_set_interrupt_wrapper` rather than `lp_ticker_set_interrupt` that gets straight called by `lp_ticker_irq_handler`. `lp_ticker_set_interrupt_wrapper` guarantees that `lp_ticker_set_interrupt` won't get re-called in `LPTICKER_DELAY_TICKS` ticks which is just enough for  "clear interrupt flag" to take effect.

This idea is implemented in https://github.com/ARMmbed/mbed-os/pull/7029/commits/26e24d604ea43a0d53fc07b96a32ef5e80da9f9b.

Reply:
@c1728p9 Any update?